packages/fxa-settings/src/lib/model-data/data-stores/url-query-data.ts

```diff
@@ -10,26 +10,75 @@ import { UrlData } from './url-data';
  * Uses window.location.search (ie the query params) to hold state.
  */
 export class UrlQueryData extends UrlData {
+  private readonly promises: Array<Promise<void>> = [];
+  private internalState: URLSearchParams;
+
   constructor(public readonly window: ReachRouterWindow) {
     super(window);
+    this.internalState = new URLSearchParams(this.window.location.search);
   }
 
+  /**
+   * Provides the current state
+   * @returns
+   */
   protected getParams() {
-    return new URLSearchParams(this.window.location.search);
+    return this.internalState;
   }
 
+  /**
+   * Sets a new internal state from a set of URL search params
+   * @param params
+   */
   protected setParams(params: URLSearchParams) {
-    const url = new URL(this.window.location.href);
-    url.search = params.toString();
-
-    // Use replace false stops a page refresh
-    this.window.navigate(url.toString(), {
-      state: this.window.location.state,
-      replace: true,
-    });
+    // Immediately update the internal state
+    this.internalState = params;
+
+    // Make a best effort to keep the url in sync. Note that
+    // this request is asynchrous which can cause weird races
+    // occaionally
```

Comment:
~asynchrous~ asynchronous 
~occaionally~ occasionally

Is this comment still accurate? I see in your commit: "Fixed an intermittent race condition in url-query-data.ts"

Reply:
I think it is still correct. It's important to call `await this.synchonized();` for this data source because of this behavior. This is taken care of internally in this class, but I think it's good context for anyone else who needs to modify or extend this class.


BTW, I just realized by cspell extension was disabled. Thanks for all the spelling corrections!